### PR TITLE
Implement pawn movements

### PR DESCRIPTION
**Pull request by pydawud**
Implementation of all basic rules of pawn piece in chess Game by dark_team.

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -52,24 +52,80 @@ def get_pawn_moves(self, r, c, moves):
...
+							if self.board[r+k][c] != "  ": # if blocked
+								pawn_attack('d')
+								break
+							else:
+								moves.append(Move((r, c), (r+k, c), self.board))
+								pawn_attack('d')
+
+					# subsequent move			
+					else:
+						if self.board[r+1][c] != "  ": # if blocked
```
Comment (drak00) [inline engine.py]: This line throws an ` IndexError: list index out of range` when dark pawn is on rank 1 (row 7)
please ensure all potential moves are within the size of the board

**Review by drak00 — CHANGES_REQUESTED** [review]

File: engine.py (hunk 1 above)
Comment (pydawud) [reply engine.py]: it is fixed now. Thanks

File: engine.py [hunk 2]
```diff
@@ -52,24 +52,82 @@ def get_pawn_moves(self, r, c, moves):
 			return parameter(s):
 			None
 		"""
+		
+		def pawn_attack(pawn_type):
```
Comment (drak00) [inline engine.py:56]: while testing, it appears that pawn_attack() always defaults to a left capture when it has both right and left pieces to capture (both light and dark pawns show this problem)

File: engine.py [hunk 3]
```diff
@@ -52,24 +52,82 @@ def get_pawn_moves(self, r, c, moves):
...
+					for i in range(-1, 2, 2):  # loops twice so j == -1 or 1 
+						if self.board[r+1][c+i][1] == 'l':  # if there's a dark piece is diagonal to it
+							return moves.append(Move((r, c), (r+1, c+i), self.board))  # create  move object and append to list			
+			
 
 		## FIX
 		if self.light_to_move: # if it's light's turn to move
 			
 			for i in range(len(self.board)):
 				for j in range(len(self.board[i])):
```
Comment (drak00) [inline engine.py]: It appears i and j are never used inside the loop! this nested loop creates the same move object and appends to list 64 times! (8 * 8) which isn't neccessary

**Review by drak00 — CHANGES_REQUESTED** [review]
Also note that your code interprets negative rows as valid, this has the unintended consequence of moving light pawns outside the board when on rank 8. To confirm this, remove all pieces on rank 1 then try moving a light pawn at rank 8.

**Review by drak00 — CHANGES_REQUESTED** [review]
Note that Light pawn still moves outside board. Kindly try fixing
Solution will be provided by tomorrow regardless

**Pull request closed without merging**